doc/api/fs.md

```diff
@@ -1313,6 +1313,19 @@ start counting at 0. If `fd` is specified and `start` is omitted or `undefined`,
 `fs.createReadStream()` reads sequentially from the current file position.
 The `encoding` can be any one of those accepted by [`Buffer`][].
 
+The following rules apply to the `start` and `end` values:
+
+- If `end` is omitted or set to undefined, then it will be coerced to `Infinity`.
+- The type of both `start` and `end` needs to be `number`. If not, then the
+function would throw a `TypeError` with the code `ERR_INVALID_ARG_TYPE`.
+- Both `start` and `end` should be valid values (i.e. they should neither be
+negative nor `NaN`, and `start` should not be greater than `end`). If they are,
+the function would throw a `RangeError` with the code `ERR_OUT_OF_RANGE`.
+- Both `start` and `end` should be integer values. If they are not, then they
+are rounded to the closest integer using `Math.round`.
+- If none of the above rules apply, then the function works without any quirks
+as expected.
```

Comment:
I’d move this closer to the end of the documentation part, since it’s mostly writing out facts that are more or less implicit in the `start` and `end ` descriptions already.

(Maybe @mcollina was rather talking about adding a `changes:` block to the metadata? That would make more sense to me than listing these conditions, tbh …)

Reply:
In that case, I must've misunderstood them. Awaiting their response, if this is not what they asked for, I'd make the required changes and remove this.

Reply:
I just generically said that this needed some documentation as it is a semver-major change.

A changes: block in the metadata should be added. Also, I would follow @addaleax suggestion.

I’d review this later (might be next week).

Reply:
Oh, from metadata you meant something like:

```yaml
<!-- YAML
added: v0.1.90
changes:
  - version: v8.8.0
    pr-url: https://github.com/nodejs/node/pull/15380
    description: The `windowsHide` option is supported now.
-->
```

right? Will make a commit doing that within the next few hours.

Reply:
@addaleax which version would this be landing in, `v10.0.0`?

Reply:
@ryzokuken The magic word for version numbers is `REPLACEME`. It’s going to land in `v11.0.0`, but please always use the former :)

Reply:
TIL. Will make the change.

Reply:
Some comments about the added documentation: 

> If `end` is omitted or set to undefined, then it will be coerced to `Infinity`.

Instead, please just write: `**Default**: Infinity` next to the argument.

> The type of both `start` and `end` needs to be `number`.

That part is obsolete because the type is already declared at the function header.

> If not, then the function would throw a `TypeError` with the code `ERR_INVALID_ARG_TYPE`. Both `start` and `end` should be valid values (i.e. they should neither be negative nor `NaN`, and `start` should not be greater than `end`). If they are, the function would throw a `RangeError` with the code `ERR_OUT_OF_RANGE`.

That should not be documented. It would be to verbose to document these things everywhere and it would be outdated super fast. Especially as most of it should be clear from the function header.

> Both `start` and `end` should be integer values. If they are not, then they are rounded to the closest integer using `Math.round`.

This can be shortened to: `If 'start' or 'end' are not integers, then they are rounded to the closest integer using 'Math.round'.`

> If none of the above rules apply, then the function works without any quirks as expected.

That should not be documented as that should always be the case ;-)

Reply:
@BridgeAR quite right. Will remove the stuff as mentioned.